torch_geometric/datasets/dgraph.py

```diff
@@ -0,0 +1,97 @@
+import os
+from typing import Callable, Optional
+
+import numpy as np
+import torch
+
+from torch_geometric.data import Data, InMemoryDataset, extract_zip
+from torch_geometric.utils import index_to_mask
+
+
+class DGraphFin(InMemoryDataset):
+    r"""The DGraphFin networks from the
+    `"DGraph: A Large-Scale Financial Dataset for Graph Anomaly Detection"
+    <https://arxiv.org/abs/2207.03579>`_ paper.
+    It is a directed, unweighted dynamic graph consisting of millions of
+    nodes and edges, representing a realistic user-to-user social network
+    in financial industry.
+    Node represents a Finvolution user, and an edge from one
+    user to another means that the user regards the other user
+    as the emergency contact person.
+
+
+    Args:
+        root (string): Root directory where the dataset should be saved.
+        transform (callable, optional): A function/transform that takes in an
+            :obj:`torch_geometric.data.Data` object and returns a transformed
+            version. The data object will be transformed before every access.
+            (default: :obj:`None`)
+        pre_transform (callable, optional): A function/transform that takes in
+            an :obj:`torch_geometric.data.Data` object and returns a
+            transformed version. The data object will be transformed before
+            being saved to disk. (default: :obj:`None`)
+
+    Stats:
+        .. list-table::
+            :widths: 10 10 10 10
+            :header-rows: 1
+
+            * - #nodes
+              - #edges
+              - #features
+              - #classes
+            * - 3,700,550
+              - 4,300,999
+              - 17
+              - 2
+    """
+
+    url = "https://dgraph.xinye.com"
+
+    def __init__(self, root: str, transform: Optional[Callable] = None,
+                 pre_transform: Optional[Callable] = None):
+        super().__init__(root, transform, pre_transform)
+        self.data, self.slices = torch.load(self.processed_paths[0])
+
+    def download(self):
```

Comment:
Any reason we don't call `download_url` here?

Reply:
See https://github.com/pyg-team/pytorch_geometric/pull/5466#issuecomment-1251990418. This dataset can only be downloaded manually by users.